src/compiler/irgen.jl

```diff
@@ -0,0 +1,455 @@
+# Julia/LLVM IR generation and transformation passes
+
+function module_setup(mod::LLVM.Module)
+    triple!(mod, "amdgcn-amd-amdhsa")
+
+    # add debug info metadata
+    push!(metadata(mod), "llvm.module.flags",
+         MDNode([ConstantInt(Int32(1), JuliaContext()),    # llvm::Module::Error
+                 MDString("Debug Info Version"),
+                 ConstantInt(DEBUG_METADATA_VERSION(), JuliaContext())]))
+end
+
+# make function names safe for PTX
```

Comment:
Does GCN have the same restrictions?

Reply:
Not sure, I'll have to go check the LLVM docs.

Reply:
Those limitations aren't documented. Try throwing some strange function names at it and see if it fails :slightly_smiling_face:

Reply:
It seems to be okay with characters that aren't even in my terminal font. `vadd_σɷ̨ℵ⌣` works as expected.